### PR TITLE
GPII-3699: Ensure that only one model change is fired for each form change.

### PR DESCRIPTION
**Pull request by the-t-in-rtf**
See [GPII-3699](https://issues.gpii.net/browse/GPII-3699) for details.

### CODE REVIEW

File: src/js/binder.js [hunk 1]
```diff
@@ -29,14 +29,15 @@
...
         var transformedValue = gpii.binder.transformPathedValue(that, path, elementValue, "domToModel");
-        var changes = [];
-        changes.push({ path: path, type: "DELETE"});
+
+        var transaction = that.applier.initiate();
+        transaction.fireChangeRequest({ path: path, type: "DELETE"});
 
         if (transformedValue !== null && transformedValue !== undefined) {
-            changes.push({ path: path, value: transformedValue });
+            transaction.fireChangeRequest({ path: path, value: transformedValue });
```
Comment (amb26) [inline src/js/binder.js:37]: Probably worthy of a test case to verify that the top-level goal is met

Comment (the-t-in-rtf) [reply src/js/binder.js:37]: Yup, done.

**Comment by the-t-in-rtf** [comment]
@amb26, this should be ready for further review.

**Comment by amb26** [comment]
npm test fails with nyc not found:

````
17:11:50.562:  FATAL ERROR: Uncaught exception: Cannot find module 'nyc'
Error: Cannot find module 'nyc'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
    at Function.Module._load (internal/modules/cjs/loader.js:506:25)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (E:\source\gits\gpii-binder\node_modules\gpii-testem\src\js\reporter.js:11:15)
````

**Comment by amb26** [comment]
Can we enrol this and other projects in CI?

**Comment by the-t-in-rtf** [comment]
Even though I usually do things the right way now, I installed nyc globally years ago.  That (and the fact that many projects include nyc as a dev dependency) kept me from spotting this up to now.  I'll work my way in this direction with a small pull against gpii-testem.

It's definitely the kind of thing CI would make crystal clear a few minutes after the original mistake was committed.  Someone still needs to come up with a CI process post-Avtar and post-Gio, I'll have to look through my notes and see how reasonable it seems to attempt it now, or if there are interim strategies such as cloning the existing jobs for each new project.

**Comment by the-t-in-rtf** [comment]
This now brings in a dev release of gpii-testem, which provides the missing nyc dependency.

**Comment by the-t-in-rtf** [comment]
Thanks, @amb26, I have cut the full release and updated the forward-facing version in master.

**Pull request merged**